core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -364,4 +376,127 @@ private static void checkCatalogType(
+                                .loadSchema(
+                                    NameIdentifier.of(
+                                        metalake,
+                                        catalogObj.name(),
+                                        "default" /*Hive default schema*/));
+                        if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+                          String defaultSchemaLocation =
+                              schema.properties().get(HiveConstants.LOCATION);
+                          if (defaultSchemaLocation != null && !defaultSchemaLocation.isEmpty()) {
+                            locations.add(defaultSchemaLocation);
+                          } else {
+                            LOG.warn("Catalog %s location is not found", ident);
+                          }
+                        }
+                      }
+                    });
+          }
+          break;
+        case CATALOG:
+          {
+            Catalog catalogObj = GravitinoEnv.getInstance().catalogDispatcher().loadCatalog(ident);
+            if (catalogObj.provider().equals("hive")) {
+              Schema schema =
+                  GravitinoEnv.getInstance()
+                      .schemaDispatcher()
+                      .loadSchema(
+                          NameIdentifier.of(
+                              metalake, catalogObj.name(), "default" /*Hive default schema*/));
+              if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+                String defaultSchemaLocation = schema.properties().get(HiveConstants.LOCATION);
+                if (defaultSchemaLocation != null && !defaultSchemaLocation.isEmpty()) {
+                  locations.add(defaultSchemaLocation);
+                } else {
+                  LOG.warn("Catalog %s location is not found", ident);
+                }
+              }
+            }
+          }
+          break;
+        case SCHEMA:
+          {
+            Schema schema = GravitinoEnv.getInstance().schemaDispatcher().loadSchema(ident);
+            if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+              String schemaLocation = schema.properties().get(HiveConstants.LOCATION);
+              if (schemaLocation != null && schemaLocation.isEmpty()) {
+                locations.add(schemaLocation);
+              } else {
+                LOG.warn("Schema %s location is not found", ident);
+              }
+            }
+          }
+          break;
+        case TABLE:
+          {
+            Table table = GravitinoEnv.getInstance().tableDispatcher().loadTable(ident);
+            if (table.properties().containsKey(HiveConstants.LOCATION)) {
+              String tableLocation = table.properties().get(HiveConstants.LOCATION);
+              if (tableLocation != null && tableLocation.isEmpty()) {
+                locations.add(tableLocation);
+              } else {
+                LOG.warn("Table %s location is not found", ident);
```

Comment:
Do you return ${schema location}/{table name} if table is null?

Reply:
The Hive table will be the return location.